etna/transforms/encoders/categorical.py

```diff
@@ -0,0 +1,228 @@
+        self.le = _LabelEncoder()
+        self.inplace = inplace
+
+    def _get_column_name(self) -> str:
+        """Get the `out_column` depending on the transform's parameters."""
+        if self.inplace and self.out_column:
+            warnings.warn("Transformation will be applied inplace, out_column param will be ignored")
+        if self.inplace:
+            return self.in_column
+        if self.out_column:
+            return self.out_column
+        if self.in_column.startswith("regressor"):
+            temp_transform = LabelEncoderTransform(
+                in_column=self.in_column, inplace=self.inplace, out_column=self.out_column, strategy=self.strategy
+            )
+            return f"regressor_{temp_transform.__repr__()}"
+        temp_transform = LabelEncoderTransform(
+            in_column=self.in_column, inplace=self.inplace, out_column=self.out_column, strategy=self.strategy
+        )
+        return temp_transform.__repr__()
+
+    def fit(self, df: pd.DataFrame) -> "_OneSegmentLabelEncoderTransform":
+        """
+        Fit Label encoder.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to fit the transform.
+        Returns
+        -------
+        self
+        """
+        self.le.fit(df[self.in_column])
+        return self
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Encode the `in_column` by fitted Label encoder.
+
+        Parameters
+        ----------
+        df
+            dataframe with data to transform.
+        Returns
+        -------
+        result dataframe
+        """
+        result_df = df.copy()
+        result_df[self._get_column_name()] = self.le.transform(df[self.in_column], self.strategy)
+        return result_df
+
+
+class LabelEncoderTransform(PerSegmentWrapper):
+    def __init__(
+        self, in_column: str, inplace: bool = True, out_column: Optional[str] = None, strategy: str = ImputerMode.mean
+    ):
+        """
+        Init LabelEncoderTransform.
+
+        Parameters
+        ----------
+        in_column:
+            name of column to be resampled
+        inplace:
+            if True, apply resampling inplace to in_column, if False, add transformed column to dataset
+        out_column:
+            name of added column. If not given, use `self.__repr__()` or `regressor_{self.__repr__()}` if it is a regressor
+        strategy:
+            filling encoding in not fitted values:
+            - If "new_value", then replace missing values with '-1'
+            - If "mean", then replace missing values using the mean in encoded column
+            - If "none", then replace missing values with None
+        """
+        self.in_column = in_column
+        self.inplace = inplace
+        self.strategy = strategy
+        self.out_column = out_column
+        super().__init__(
+            transform=_OneSegmentLabelEncoderTransform(self.in_column, self.out_column, self.strategy, self.inplace)
```

Comment:
Use keyword parameters here like in_column=self.in_column.
This can be a reason for warning during creation with out_column=None.